sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveDirCommand.scala

```diff
@@ -90,6 +90,15 @@ case class InsertIntoHiveDirCommand(
         val dfs = qualifiedPath.getFileSystem(hadoopConf)
         (qualifiedPath, dfs)
       }
+
+    if (fs.exists(writeToPath) && fs.isFile(writeToPath)) {
+      if (overwrite) {
+        fs.delete(writeToPath, true)
+      } else {
+        throw new SparkException(s"Failed inserting overwrite directory, " +
```

Comment:
What's the current behaivour when overwrite=false?

Reply:
There is currently only insert overwrite directory syntax, overwrite must be true now, the code retains this parameter, I feel that the designer for future expansion.

Reply:
Ah, I see. Even so, we need tests for this path.